### ISSUE TITLE
Always respect agentName

### ISSUE THREAD

**Issue by ahdung** [Ylianst/MeshCentral#5601]
Adding a option "Sync device name from agentName" for Group, please.

**Comment by silversword411** [comment]
Already there. Edit group features
![2023-12-02_065725 - mesh sync name](https://github.com/Ylianst/MeshCentral/assets/13395348/392422e5-a51f-44d2-a74e-8ffb7de5fecc)
Please close

**Comment by si458** [comment]
I think they are referring to the agentname set inside the msh file?
https://github.com/Ylianst/MeshAgent#msh-format

**Comment by ahdung** [comment]
@si458 I am referring to the `--agentName="xxx"` parameter during installation. I hope allow rename it(I modify the ImagePath of service), and device name follow it.

**Comment by ahdung** [comment]
@si458 `--agentName` not write in any configuration file, it is a part of the `Mesh Agent` service's command line like this:
```
"C:\Program Files\Mesh Agent\MeshAgent.exe" --agentName="C爱的色放" --installedByUser="S-1-5-21-427675387-1404930041-4285012813-500"
```

This is fine, I just hope can rename it without reinstall, and device name on server side auto follow it.

**Comment by si458** [comment]
@ahdung ive just tried it here and seems ok, can you confirm this is how you set it up?
(NOTE: i customised my agent for testing so change `myremoteagent` to `meshagent` if you havent customised it)

1. installed using command line
`C:\users\administrator\downloads\myremoteagen64-testgroup.exe -fullinstall --agentName="billybob"`
2. shows up in webpanel with the agentName `billybob`  ?

**Comment by ahdung** [comment]
@si458 First of all thank you for your patient help. I think you misunderstood my question, I didn't mean the `--agentName` doesn't work, and it also support non-ascii, so it is fine, no problem. My requestion is this name should be rename after installation, and the device name  will update to new name, The focus is on the **rename**.

So my request has 2 points:
- Agent installer provide a new parameter for rename, maybe like `-rename=xxx`. This is not necessary, I can write wrapper to do it.
- Device name auto change to "xxx".

**Comment by si458** [comment]
Ah... yes I noticed this before!
The only way around this at the moment, is to simply remove the device from the webui, then run the install command again with the agentname set or set agentname inside ur msh file which gets created when u do the fullinstall, then restart the agent on the remote device!
Hope this helps for the moment!

**Comment by ahdung** [comment]
Yep, this is exactly how I do it currently, I just want it to be more elegant and not require workarounds.

**Comment by si458** [comment]
this was already added i believe - https://github.com/Ylianst/MeshCentral/pull/6809
so closing as duplicate of https://github.com/Ylianst/MeshCentral/issues/6808
